src/braket/circuits/circuit.py

```diff
@@ -438,6 +440,136 @@ def add_circuit(
 
         return self
 
+    def add_noise(
+        self,
+        noise: Noise,
+        target_gates: Union[str, Iterable[str]] = None,
+        target_qubits: QubitSetInput = None,
+        target_times: Union[int, Iterable[int]] = None,
+        insert_strategy: str = "inclusive",
+    ) -> Circuit:
+        """
+        Add the provided `noise` to gates, qubits and time specified by `target_gates`,
+         `target_qubits` and `target_times`. When `target_gates`=None, `noise` is added
+        to the specified qubits and time. When `target_qubits`=None, `noise` is added to
+        the specified gates and time at all qubits. When `target_times`=None, `noise` is
+        added to the specified gates and qubits at all time. If all `target_gates`,
+        `target_qubits` and `target_times` are None, `noise` is added to every qubit at
+        every time in a circuit.
+
+        When `target_gates` is not None and `noise.qubit_count`>1, `noise.qubit_count`
+        must be the same as `qubit_count` of gates specified by `target_gates`. When
+        `noise.qubit_count`==1, ie. `noise` is a single-qubit noise, `noise` is added
+        to all qubits in `target_qubits` that interact with the target gates.
+
+        Args:
+            noise (Noise): Noise to be added to the circuit. When `noise.qubit_count`>1,
+                `noise.qubit_count` must be the same as `qubit_count` of gates specified by
+                `target_gates`.
+            targer_gates (Union[str, Iterable[str]): Name or List of name of gates which
```

Comment:
For the existing code, users can specify the time and qubit a specific unitary locates. 
The Gate objects do not have unique name. So it is not straightforward to compare to gates in general. For example, when comparing two Rx gates, we can compare their angle. But for for differently parametrized gates, we might want to compare something else. We could always compare the equivalence of matrices, but not sure that leads to the best customer experience because defining a matrix is a bit cumbersome. There are a few other options, but I don't think we  have enough data to determine which behavior fits customer needs the best. So I just kept it simple, and only allow specifying a specific gate through gate name, time and qubit. But we can always add new functionality later.